homework5/main5.js

```diff
@@ -0,0 +1,271 @@
+  firstArr[i] = firstArr[i][0].toUpperCase() + firstArr [i].slice(1);
+}
+let newStr = firstArr.join('');
+handler1 (newStr);
+
+}
+firstFunc(['my', 'name', 'is', 'Trinity'], function (num) {
+    console.log('New value:' + num)
+})
+
+
+// firstFunc([10, 20, 30], handler2) → “New value: 100, 200, 300,”
+
+function firstFunc(firstArr, handler1){
+
+    for (let i = 0; i < firstArr.length; i++) {
+        firstArr[i] *=10;
+    }
+    handler1 (firstArr);
+
+}
+firstFunc([10, 20, 30], function (num) {
+    console.log('New value:' + num)
+})
+
+
+// firstFunc([{age: 45, name: ‘Jhon’}, {age: 20, name: ‘Aaron’}], handler3) →
+// “New value: Jhon is 45, Aaron is 20,”
+
+function firstFunc(firstArr, firstValue, secondValue, handler3){
+    var output = [];
+
+
+    for (var i=0; i < firstArr.length ; ++i) {
+        output.push(firstArr[i][firstValue], ' ', 'is', ' ', firstArr[i][secondValue], ' ')
+    }
+    var secondStr = output.join('');
+    handler3 (secondStr);
+
+}
+firstFunc([{age: 45, name: 'John'}, {age: 20, name: 'Aaron'}], 'name','age',function (num) {
+    console.log('New value:' + num)
+})
+
+//firstFunc([‘abc’, ‘123’], handler4) → “New value: cba, 321,” // строки инвертируются
+
+function firstFunc(firstArr, handler1){
+    var firstString = firstArr.join(' ')
+    var newString = "";
+    for (var i = firstString.length-1; i>=0; i--){
+        newString+= firstString[i];
+    }
+    let secondString = newString.split(' ');
+    secondString.reverse()
+
+    handler1 (secondString);
+
+}
+firstFunc(['abc', '123'], function (num) {
+    console.log('New value:' + num)
+})
+
+// 2. Написать аналог метода every. Создайте функцию every, она должна принимать первым
+// аргументом массив чисел (обязательно проверьте что передан массив) вторым аргументом
+// callback
+// функция должна возвращать true или false в зависимости от результата вызова callback (проверить
+// число больше 5). Callback должен принимать один элемент массива, его индекс в массиве и весь
+// массив.
+
+function firstFunc(numberArr, callback){
+
+for (i=0; i < numberArr.length; i++){
+    var index = numberArr[i].length;
+    var elem = numberArr[i];
+    if(elem > 5){
+        return true
+    } else {
+        return false
+    }
+    }
+
+    callback (elem, index, numberArr);
+
+}
+firstFunc([6,8], function (num) {
+    console.log('New value:' + num)
+})
+
+// 1. На основе массива [1,2,3,5,8,9,10] сформировать новый массив,
+//     каждый элемент которого будет хранить информацию о числе и его четности:
+//     [{digit: 1, odd: true}, {digit: 2, odd: false}, {digit: 3, odd: true}...]
+
+
+var arr = [1,2,3,5,8,9,10];
+var result = {};
+
+var newArr = arr.map(function(elem, index) {
+    if(elem % 2 > 0){
+        result = {
+            digit: index,
+            odd:'true'}
+        return arr[elem] = result;
+    } else {
+        result = {
+            digit: index,
+            odd:'false'}
+        return arr[elem] = result;
```

Comment:
You got wrong `map` method. You should have returned just object you needed. Thats all.
`return {`
`digit: index,`
`odd: false`
`}`
There is no need to do `arr[elem] = result`